### ISSUE TITLE
Photo downloads fail in TB68

### ISSUE THREAD

**Issue by jdgeenen** [jdgeenen/gcontactsync#136]
**Describe the bug**
Photo downloads fail in TB68
```
TypeError: ios.newChannel2 is not a function misc.js:678:15
    gCS_createChannel chrome://gcontactsync/content/misc.js:678
    GContact_writePhoto chrome://gcontactsync/content/GContact.js:955
    Sync_processDownloadPhotoQueue chrome://gcontactsync/content/Sync.js:898
    Sync_processUpdatePhotoQueue chrome://gcontactsync/content/Sync.js:857
    Sync_processUpdateQueue chrome://gcontactsync/content/Sync.js:792
    Sync_processAddQueue chrome://gcontactsync/content/Sync.js:721
    Sync_processDeleteQueue chrome://gcontactsync/content/Sync.js:681
    syncWorkerOnMessage chrome://gcontactsync/content/Sync.js:622
    onmessage chrome://gcontactsync/content/Sync.js:560
```

**To Reproduce**
1. Add a contact w/ a photo from google.com/contacts
2. Sync.  Photo missing, sync doesn't finish.
3. Open the Browser Console

**Expected behavior**
Photo downloaded with no errors

**Screenshots**

**Application (please complete the following information):**
 - Application: Thunderbird
 - Application Version: 68
 - gContactSync Version: 3.1.15

**Additional context**
latest nightly as of 6/1

**Comment by jdgeenen** [comment]
Replacing newChannel2 with newChannel causes this:
```NS_ERROR_FAILURE: Component returned failure code: 0x80004005 (NS_ERROR_FAILURE) 
[nsIChannel.open] GContact.js:958
    GContact_writePhoto chrome://gcontactsync/content/GContact.js:958
    Sync_processDownloadPhotoQueue chrome://gcontactsync/content/Sync.js:898
    Sync_processUpdatePhotoQueue chrome://gcontactsync/content/Sync.js:857
    Sync_processUpdateQueue chrome://gcontactsync/content/Sync.js:792
    Sync_processAddQueue chrome://gcontactsync/content/Sync.js:721
    Sync_processDeleteQueue chrome://gcontactsync/content/Sync.js:681
    syncWorkerOnMessage chrome://gcontactsync/content/Sync.js:622
    onmessage chrome://gcontactsync/content/Sync.js:560
```
Setting the load info and policy fixed this:
                         Ci.nsILoadInfo.SEC_ALLOW_CROSS_ORIGIN_DATA_INHERITS,
                         Ci.TYPE_OBJECT);

**Comment by jdgeenen** [comment]
Photo uploads work with my changes, too

**Comment by jdgeenen** [comment]
This should be fixed in 3.1.6.
